src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -355,18 +355,23 @@ internal virtual void GetExtensionMethods(ArrayBuilder<MethodSymbol> methods, st
             }
         }
 
-        internal void GetExtensionContainers(ArrayBuilder<NamedTypeSymbol> extensions)
+        internal virtual void GetExtensionContainers(ArrayBuilder<NamedTypeSymbol> extensions)
         {
             foreach (var type in this.GetTypeMembersUnordered())
             {
-                if (!type.IsReferenceType || !type.IsStatic || type.IsGenericType || !type.MightContainExtensionMethods) continue;
+                AddExtensionContainersForType(type, extensions);
+            }
+        }
+
+        protected void AddExtensionContainersForType(NamedTypeSymbol type, ArrayBuilder<NamedTypeSymbol> extensions)
```

Comment:
"AddExtensionContainersFromType" or "AddExtensionContainersInType"?

Reply:
Done in commit 4